lib/thor.rb

```diff
@@ -238,6 +238,9 @@ def subcommand(subcommand, subcommand_class)
         args.unshift("help") if opts.include? "--help" or opts.include? "-h"
         invoke subcommand_class, args, opts, :invoked_via_subcommand => true, :class_options => options
       end
+      subcommand_class.commands.each do |meth, command|
```

Comment:
What would be the downside of simply setting the namespace on the class to the subcommand name by default ?